receiver/mongodbatlasreceiver/logs.go

```diff
@@ -0,0 +1,306 @@
+}
+
+type ProjectContext struct {
+	Project   mongodbatlas.Project
+	orgName   string
+	projectID string
+	start     time.Time
+	end       time.Time
+}
+
+// MongoDB Atlas Documentation reccommends a polling interval of 5  minutes: https://www.mongodb.com/docs/atlas/reference/api/logs/#logs
+const collectionInterval = time.Minute * 5
+
+func newMongoDBAtlasLogsReceiver(settings component.ReceiverCreateSettings, cfg *Config, consumer consumer.Logs) (*logsReceiver, error) {
+	client, err := internal.NewMongoDBAtlasClient(cfg.PublicKey, cfg.PrivateKey, cfg.RetrySettings, settings.Logger)
+	if err != nil {
+		return nil, err
+	}
+	return &logsReceiver{
+		log:         settings.Logger,
+		cfg:         cfg,
+		client:      client,
+		stopperChan: make(chan struct{}),
+		consumer:    consumer}, nil
+}
+
+// Log receiver logic
+func (s *logsReceiver) Start(ctx context.Context, host component.Host) error {
+	s.wg.Add(1)
+	go func() {
+		defer s.wg.Done()
+		start := time.Now()
+		for {
+			// collection interval loop,
+			select {
+			case <-ctx.Done():
+				return
+			case <-s.stopperChan:
+				return
+			case <-time.After(collectionInterval):
+			}
+			start = s.collect(ctx, start)
+		}
+	}()
+	return nil
+}
+
+func (s *logsReceiver) Shutdown(ctx context.Context) error {
+	close(s.stopperChan)
+	s.wg.Wait()
+	return s.client.Shutdown()
+}
+
+// parseHostNames parses out the hostname from the specified cluster host
+func parseHostNames(s string, logger *zap.Logger) []string {
+	var hostnames []string
+
+	if s == "" {
+		return []string{}
+	}
+
+	for _, t := range strings.Split(s, ",") {
+		// separate hostname from scheme and port
+		host, _, err := net.SplitHostPort(strings.TrimPrefix(t, "mongodb://"))
+		if err != nil {
+			logger.Error("Could not parse out hostname: " + host)
+			continue
+		}
+		hostnames = append(hostnames, host)
+	}
+
+	return hostnames
+}
+
+// collect spins off functionality of the receiver from the Start function
+func (s *logsReceiver) collect(ctx context.Context, start time.Time) time.Time {
+	pc := ProjectContext{start: start, end: time.Now()}
+	for _, projectCfg := range s.cfg.Logs.Projects {
+		project, err := s.client.GetProject(ctx, projectCfg.Name)
+		if err != nil {
+			s.log.Error("Error retrieving project "+projectCfg.Name+":", zap.Error(err))
+			continue
+		}
+
+		org, err := s.client.GetOrganization(ctx, project.OrgID)
+		if err != nil {
+			s.log.Error("Error retrieving organization", zap.Error(err))
+			pc.orgName = "unknown"
+		} else {
+			pc.orgName = org.Name
+		}
+
+		// get clusters for each of the projects
+		pc.Project = *project
+		clusters, err := s.processClusters(ctx, *projectCfg, project.ID)
+		if err != nil {
+			s.log.Error("Failure to process Clusters", zap.Error(err))
+		}
+
+		s.collectClusterLogs(clusters, *projectCfg, pc)
+	}
+	return pc.end
+}
+
+func (s *logsReceiver) processClusters(ctx context.Context, project ProjectConfig, projectID string) ([]mongodbatlas.Cluster, error) {
```

Comment:
`project` is overloaded between functions. Technically it's ok, but it makes it difficult to follow.
```suggestion
func (s *logsReceiver) processClusters(ctx context.Context, projectCfg ProjectConfig, projectID string) ([]mongodbatlas.Cluster, error) {
```